src/commands/update.ts

```diff
@@ -0,0 +1,68 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import { bin_name, config } from '..'
+import { log } from '../log'
+
+import {
+  setupFirefoxSource,
+  shouldSetupFirefoxSource,
+} from './download/firefox'
+import {
+  addAddonsToMozBuild,
+  downloadAddon,
+  generateAddonMozBuild,
+  getAddons,
+  initializeAddon,
+  resolveAddonDownloadUrl,
+  unpackAddon,
+} from './download/addon'
+import { getLatestFF } from '../utils'
+import { rmSync } from 'node:fs'
+import { ENGINE_DIR } from '../constants'
+
+export const update = async (): Promise<void> => {
+  const version = await getLatestFF(config.version.product)
+
+  // If gFFVersion isn't specified, provide legible error
+  if (!version) {
+    log.error(
+      'You have not specified a version of firefox in your config file. This is required to build a firefox fork.'
+    )
+    process.exit(1)
+  }
+
+  if (!shouldSetupFirefoxSource()) {
```

Comment:
This seems like a bit of duplication with the download internals. I would prefer if the internals of download that you need were split off into a different function that is called in both places

Reply:
It is a duplication

Reply:
I'm not sure what you mean by that, could you elaborate?

Reply:
I presume the reason you can't just call the the download function in `download.ts` is because:
a) The version at the top is incorrect
b) The print statement at the bottom is incorrect

What I would do is create a new function like this:
```ts
export async function donwloadInternals(version: string) {
  // If gFFVersion isn't specified, provide legible error
  if (!version) {
    log.error(
      'You have not specified a version of firefox in your config file. This is required to build a firefox fork'
    )
    process.exit(1)
  }

  if (shouldSetupFirefoxSource()) {
    await setupFirefoxSource(version)
  }

  for (const addon of getAddons()) {
    const downloadUrl = await resolveAddonDownloadUrl(addon)
    const downloadedXPI = await downloadAddon(downloadUrl, addon)

    await unpackAddon(downloadedXPI, addon)
    await generateAddonMozBuild(addon)
    await initializeAddon(addon)
  }

  await addAddonsToMozBuild(getAddons())
}
```

And call it in both locations. i.e. 
```ts
export const update = async (): Promise<void> => {
  const version = await getLatestFF(config.version.product)

  await donwloadInternals(version)

  log.success(
    `Firefox has sucessfully been updated to ${version}.`,
    `You should be ready to make changes to ${config.name}.`,
    '',
    `You should import the patches next, run |${bin_name} import|.`,
    `To begin building ${config.name}, run |${bin_name} build|.`
  )
  console.log()
}
```

Reply:
Should be resolved